examples/ensemble/plot_adaboost_hastie_10_2.py

```diff
@@ -15,29 +15,39 @@
 .. [1] T. Hastie, R. Tibshirani and J. Friedman, "Elements of Statistical
     Learning Ed. 2", Springer, 2009.
 
-.. [2] J. Zhu, H. Zou, S. Rosset, T. Hastie, "Multi-class AdaBoost", 2009.
+.. [2] J Zhu, H. Zou, S. Rosset, T. Hastie, "Multi-class AdaBoost",
+    Statistics and Its Interface, 2009.
 
 """
 
-# Author: Peter Prettenhofer <peter.prettenhofer@gmail.com>,
-#         Noel Dawe <noel.dawe@gmail.com>
+# %%
+# Preparing the data and baseline models
+# --------------------------------------
+# We start by generating the binary classification dataset
+# used in Hastie et al. 2009, Example 10.2.
+
+# Authors: Peter Prettenhofer <peter.prettenhofer@gmail.com>,
+#          Noel Dawe <noel.dawe@gmail.com>
 #
 # License: BSD 3 clause
 
-import numpy as np
-import matplotlib.pyplot as plt
-
 from sklearn import datasets
-from sklearn.tree import DecisionTreeClassifier
-from sklearn.metrics import zero_one_loss
-from sklearn.ensemble import AdaBoostClassifier
 
+X, y = datasets.make_hastie_10_2(n_samples=12000, random_state=1)
```

Comment:
```suggestion
from sklearn.model_selection import train_test_split

X, y = datasets.make_hastie_10_2(n_samples=12_000, random_state=1)
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=2_000, shuffle=False
)
```

Reply:
I moved the train_test_split to the actual split part down below though. The loading of the dataset remains in its own cell. I actually like the flow of that a bit better for the example.